src/pages/task/testsTable/StatusSelector.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { useLocation, useHistory } from "react-router-dom";
+import queryString from "query-string";
+import { RequiredQueryParams } from "types/task";
+import { TestStatus } from "types/task";
+import { TreeSelect } from "components/TreeSelect";
+
+export const StatusSelector = () => {
+  const { pathname, search } = useLocation();
+  const { replace } = useHistory();
+  const value = useStatuses(search);
+
+  const onChange = (updatedValue: [string]) => {
+    const parsed = queryString.parse(search, { arrayFormat });
+    parsed[RequiredQueryParams.Statuses] = updatedValue;
+    const nextQueryParams = queryString.stringify(parsed, { arrayFormat });
+    replace(`${pathname}?${nextQueryParams}`);
+  };
+
+  return (
+    <TreeSelect
+      onChange={onChange}
+      state={value}
+      tData={treeData}
+      inputLabel="Test Status:  "
+      id="cy-test-status-select"
+    />
+  );
+};
+
+const arrayFormat = "comma";
+
+const treeData = [
```

Comment:
+1 to putting these things below the component so that the component is the first code in the file.